src/modules/claims/claims.service.ts

```diff
@@ -1421,22 +1424,67 @@ export class ClaimsService {
     await this._vcIssuerVerifier.verifyIssuer(issuerDID, role);
   }
 
+  /**
+   * Verifies that off-chain claim has valid proof and was issued by an authorized issuer
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   * @return void. Returns "Invalid issuer for off-chain claim" error if issuer not verified. Returns 'The credential is invalid' error if invalid proof
+   */
+  async verifyOffChainClaim(subjectDid: string, roleNameSpace: string): Promise<void> {
+    const issuerDID = this._signerService.did;
+    const claimVerifier = new ClaimIssuerVerification(this._signerService.provider, this._didRegistry.registrySettings, this._credentialResolver, this._issuerResolver);
+    if (!(await claimVerifier.verifyIssuer(issuerDID, roleNameSpace))) {
+      throw new Error(ERROR_MESSAGES.INVALID_ISSUER_OFFCHAIN_CLAIM)
+    }
+    await claimVerifier.verifyIssuance(subjectDid, roleNameSpace)
+  }
+
+  /**
+   * Determines the verification method for a claim
+   *
+   * @param {VerifiableCredential<RoleCredentialSubject} vc to be verified
+   * @return void. Returns "Proof Not Verified" error if VC not verified. Returns error if issuer not verified
+   */
+  async verify(subjectDID: string, roleNamespace: string): Promise<void> {
+    try {
+      /*
+       This will not return a VC based on our current implementation, which requires a VC URL.
```

Comment:
@nichonien sorry if I'm missing this, but I don't get why we would need the try catch here? The getCredential will return undefined if no vc is found, it won't throw an error. so I don't think we need to put the logic for the offchain fetch in the catch logic? I might have misunderstood what you said. 

@jrhender @JGiter would like feedback if this logic encapsulates everything from our discussion yesterday on the approach.

Reply:
Oh great, thanks @nichonien ! I will update ICL to use this new method then. Thanks